api/src/main/java/org/apache/iceberg/IncrementalAppendScan.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.iceberg;
+
+/**
+ * API for configuring an incremental table scan for appends only snapshots
+ */
+public interface IncrementalAppendScan extends Scan<IncrementalAppendScan> {
+
+  /**
+   * Refine the incremental scan with the start snapshot inclusive.
+   * <p>
+   * If the start snapshot (inclusive or exclusive) is not provided,
+   * the oldest ancestor of the {@link IncrementalAppendScan#toSnapshot(long)}
+   * will be included as the start snapshot.
```

Comment:
@stevenzwu, this isn't true. If the starting snapshot is not set, then it defaults to `null`, which will scan from the start of table history or fail if table history has expired.

Reply:
@rdblue yes, it defaults to `null`. The described behavior is based on `IncrementalDataTableScan` using `SnapshotUtil.ancestorsBetween`, which scan from the oldest ancestor of the `toSnapshotId`. I think the current behavior makes sense. Start of table history may not be an ancestor of the `toSnapshotId`, right?

Reply:
Maybe the description can be simplified as `if ..., this scans all the ancestors of the {@link IncrementalAppendScan#toSnapshot(long)}`

Reply:
I don't think the proposed behavior makes sense for this interface. Otherwise, there is no way to incrementally scan from the start of the table. If I want to start from the beginning of history, I need to specify starting snapshot `null`. But there's no way to do that without leaving out the "from" snapshot. If that's how to configure scanning from the start of history, then this can't scan from the oldest known snapshot by default.

Reply:
if the oldest table snapshot is not an ancestor of the `toSnapshot`, what does it mean? I thought incremental scan is only meaningful along the linear ancestor line.

E.g., we have two disjointed lineages. 
S1 -> S2     ---------->     S5
               S3 -> S4 -> S6 -> S7

If the `toSnapshotId` is set to `S7` and `fromSnapshotId` is not set, I thought we want to scan [S3, S4, S6, S7]. Is that correct?

Reply:
The problem is when the history has expired, not when there is no ancestor relationship. When the starting point is not an ancestor, that's a different problem that results in an exception.

Reply:
Just to make sure I understand you correctly. if `fromSnapshotId` is not set and defaults to null, we want start from the snapshot with the oldest timestamp using this `Table` API: `Iterable<Snapshot> snapshots()`. if the oldest snapshot is not an ancestor of the current table snapshot, we throw an exception.

I assume we don't want to use this `Table` API to find out the oldest snapshot (by timestamp)
```
List<HistoryEntry> history()
```

Reply:
If no `fromSnapshot` method is called, the incremental read should start from the beginning of table history, the `null` snapshot. So we need to find a snapshot with `parent-snapshot-id=null`.

Reply:
if the table history has disjointed lineage lines, then we can have multiple snapshots with `parent-snapshot-id=null`. I guess then we can use timestamp to break the tie.

Reply:
@rdblue I updated the Javadoc based on the discussion here.

Reply:
@rdblue can you take another look and see if the comments are addressed adequately.